src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java

```diff
@@ -165,32 +169,28 @@ private boolean isLatestRecord(String officerId, OffsetDateTime deltaAt) {
      */
     private void saveAndCallChsKafka(String contextId, String officerId,
             DisqualificationDocument document, DisqualificationResourceType type) {
-        boolean savedToDb = false;
         Created created = getCreatedFromCurrentRecord(officerId);
         if(created == null) {
             document.setCreated(new Created().setAt(document.getUpdated().getAt()));
         } else {
             document.setCreated(created);
         }
 
+        disqualifiedOfficerApiService.invokeChsKafkaApi(
+                new ResourceChangedRequest(contextId, officerId,
+                        type, null, false));
+        logger.info(String.format("ChsKafka api CHANGED invoked updated successfully for context id: %s and officer id: %s",
+                contextId,
+                officerId));
+
         try {
             repository.save(document);
-            savedToDb = true;
             logger.info(String.format("Disqualification is updated in MongoDb for context id: %s and officer id: %s",
-                contextId,
-                officerId));
+                    contextId,
+                    officerId));
```

Comment:
Line 204 could be updated to a more functional style:
`return doc.map(DisqualificationDocument::getCreated).orElse(null);`

Line 210 can be returned inline:
`return disqualificationDocumentOptional.orElseThrow(
                () -> new IllegalArgumentException(String.format(
                "Resource not found for officer ID: %s", officerId)));`

Reply:
Fixed in latest commit.